### ISSUE TITLE
lua api: Function to get adjacent nodes.

### ISSUE THREAD

**Issue by hi563145** [minetest/minetest#14110]
### Problem

I think it would be useful to have a Lua API to check for adjacent nodes of a given location. This is different to `find_node_near` as the blocks found would be directly touching the node where the `find_node_near` function will get nodes that are diagonal to it.

### Solutions

Adds a new Lua API function to find touching blocks.

### Alternatives

Add an option to `find_node_near` or `find_nodes_in_area` to only search for touching blocks.

**Comment by sfan5** [comment]
Rather than hardcoding a potential new function to certain neighbors I suggest implementing `bulk_get_node`.
Just like `bulk_set_node` it would be faster and more convenient than individual calls and also still faster than using a VoxelManip (due to setup costs).

**Comment by Desour** [comment]
As `bulk_get_node` is rejected as API (at least if there's no performance gain) (for now), "concept approved" no longer holds.

Do we want something like a `find_touching_nodes` function or similar?
(Imo not, modders should implement their own helper functions.)

**Comment by appgurueu** [comment]
> Imo not, modders should implement their own helper functions.

Agreed, that would IMO be too specific an API for us to support. Maybe there are performance reasons for such an API, but that would first have to be demonstrated.
